### PR TITLE
Increase ep_bulk_items_per_page and EP_SYNC_CHUNK_LIMIT to 500

### PR DESCRIPTION
**Pull request by brandon-m-skinner**
## Description

PLAT-885

Increase ep_bulk_items_per_page and EP_SYNC_CHUNK_LIMIT to 500

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

TBD

### CODE REVIEW

File: elasticsearch/elasticpress/includes/classes/Indexable.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ public function get_bulk_items_per_page() {
 		 * @return  {int} New number of items
 		 * @since  3.0
 		 */
-		return apply_filters( 'ep_bulk_items_per_page', 350, $this );
+		return apply_filters( 'ep_bulk_items_per_page', 500, $this );
```
[the remark this answers is not in the document]
Comment (brandon-m-skinner) [reply elasticsearch/elasticpress/includes/classes/Indexable.php]: Fair enough. The only thing not in the sub-module is the test, so back to the drawing board.

**Comment by brandon-m-skinner** [comment]
The PR is still missing the `per_page` setting. Unless the setting of the `per_page` doesn't matter? I'll just add this in case it does still matter:

Does it make sense to not bother with` ep_bulk_items_per_page` and instead simply set the `per_page` key to 500 in `query_args` in the health check code to satisfy the:

> The ep_bulk_items_per_page filter controls how many posts are indexed at once during bulk indexing. It defaults to 350, but in my testing 500 is the sweet spot for max throughput.

requirement in PLAT-885? Probably just make a separate constant unless just using `EP_SYNC_CHUNK_LIMIT` here makes sense well if they're always going to be the same?

I've been combing through everything and the filter only seems to be used as a default if the query_args['per_page'] isn't set.

Without doing this, the per_page would still be 350 while the EP_SYNC_CHUNK_LIMIT will be 500.

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 2]
```diff
@@ -50,6 +50,9 @@ protected function setup_constants() {
 	protected function setup_hooks() {
 		add_filter( 'ep_index_name', [ $this, 'filter__ep_index_name' ], PHP_INT_MAX, 3 ); // We want to enforce the naming, so run this really late.
 
+		// Override default per page value set in elasticsearch/elasticpress/includes/classes/Indexable.php
+		add_filter( 'ep_bulk_items_per_page', function() { return 500; } );
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-elasticsearch.php]: To keep this aligned with the other filters we have, we should switch this out to a function on the class (I mostly suggested an anonymous function here as a PoC).

File: elasticsearch/includes/classes/class-elasticsearch.php [hunk 3]
```diff
@@ -50,6 +50,9 @@ protected function setup_constants() {
 	protected function setup_hooks() {
 		add_filter( 'ep_index_name', [ $this, 'filter__ep_index_name' ], PHP_INT_MAX, 3 ); // We want to enforce the naming, so run this really late.
 
+		// Override default per page value set in elasticsearch/elasticpress/includes/classes/Indexable.php
+		add_filter( 'ep_bulk_items_per_page', [ $this, 'filter__ep_bulk_items_per_page' ], PHP_INT_MAX, 0 );
```
Comment (nickdaugherty) [inline elasticsearch/includes/classes/class-elasticsearch.php]:
```suggestion
		add_filter( 'ep_bulk_items_per_page', [ $this, 'filter__ep_bulk_items_per_page' ], PHP_INT_MAX );
```

The "args count" param can be omitted here - it's only needed if your callback is expecting more than 1 argument.

**Comment by nickdaugherty** [comment]
r768-stacks

**Pull request merged**